src/WormholeL2GovernorMetadata.sol

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.16;
+
+import {WormholeReceiver} from "src/WormholeReceiver.sol";
+import {WormholeBase} from "src/WormholeBase.sol";
+import {L2GovernorMetadata} from "src/L2GovernorMetadata.sol";
+
+/// @notice Use Wormhole to receive L1 proposal metadata.
+contract WormholeL2GovernorMetadata is L2GovernorMetadata, WormholeReceiver {
+  /// @param _relayer The address of thWormholeL2GovernorMetadata contract.
+  constructor(address _relayer) WormholeBase(_relayer) L2GovernorMetadata() {}
+
+  /// @notice Receives a message from L1 and saves the proposal metadata.
+  /// @param payload The payload that was sent to in the delivery request.
+  function receiveWormholeMessages(bytes memory payload, bytes[] memory, bytes32, uint16, bytes32)
+    public
+    override
+    onlyRelayer
+  {
+    // TODO: verify message's sourceAddress (and sourceChain?)
```

Comment:
This will be handled with a modifier in #37

Reply:
This can probably be removed

Reply:
removed in 696ed96